bin/certificates.py

```diff
@@ -149,6 +149,11 @@ def construct_template_path(root_dir, badge_type):
 def construct_output_path(root_dir, badge_type, user_id):
     '''Create path for generated PDF certificate.'''
 
+    path_exists = os.path.exists(os.path.join(root_dir, badge_type))
+
+    if not path_exists:
+        os.mkdir(os.path.join(root_dir, badge_type))
+
     return os.path.join(root_dir, badge_type, user_id + '.pdf')
```

Comment:
I would suggest this to reduce duplication and operations

```suggestion
    badge_path = os.path.join(root_dir, badge_type)
    path_exists = os.path.exists(badge_path)

    if not path_exists:
        os.mkdir(badge_path)

    return os.path.join(badge_path, user_id + '.pdf')
```